src/controllers/utils/minimizeWords.ts

```diff
@@ -1,29 +1,47 @@
 import { assign, omit, pick } from 'lodash';
+import { Types } from 'mongoose';
 import Version from '../../shared/constants/Version';
+import { Definition, LegacyWordDialect, PartialWordType, WordDialect } from '../../types/word';
+import { Example } from '../../types';
 
-const minimizeWords = (words, version) => {
+type MinimizedWord = Omit<PartialWordType, 'definitions' | 'examples' | 'dialects' | 'relatedTerms' | 'stems'> & {
+  definitions: Partial<Definition>[] | string[] | undefined;
+  examples: Partial<Example>[];
+  tenses: string[] | undefined;
+  dialects?: Partial<WordDialect>[] | LegacyWordDialect | undefined;
+  relatedTerms?: (string | Partial<{ id: string; _id?: Types.ObjectId }>)[];
+  stems?: (string | Partial<{ id: string; _id?: Types.ObjectId }>)[];
+};
+const minimizeWords = (words: PartialWordType[], version: Version) => {
   console.time('Minimize words');
   const minimizedWords = words.map((word) => {
-    let minimizedWord = assign(word);
+    let minimizedWord: Partial<MinimizedWord> = assign(word);
     minimizedWord = omit(minimizedWord, ['hypernyms', 'hyponyms', 'updatedAt', 'createdAt']);
     minimizedWord.definitions =
       version === Version.VERSION_2
-        ? (minimizedWord.definitions || []).map((definition) => {
-            let minimizedDefinition = assign(definition);
-            minimizedDefinition = omit(minimizedDefinition, ['label', 'igboDefinitions', '_id', 'id']);
-            if (!minimizedDefinition.nsibidi) {
-              minimizedDefinition = omit(minimizedDefinition, ['nsibidi']);
+        ? ((minimizedWord.definitions || []).map((definition) => {
+            let minimizedDefinition: Partial<Definition> | string = assign(definition);
+            if (typeof definition === 'object') {
+              minimizedDefinition = omit(minimizedDefinition as Partial<Definition>, [
+                'label',
+                'igboDefinitions',
+                '_id',
+                'id',
+              ]);
+              if (!minimizedDefinition.nsibidi) {
+                minimizedDefinition = omit(minimizedDefinition, ['nsibidi']);
+              }
             }
             return minimizedDefinition;
-          })
-        : minimizedWord.definitions;
+          }) as Partial<Definition>[] | string[])
+        : (minimizedWord.definitions as undefined);
```

Comment:
why are you casting `definitions` as `undefined`? it's not expected that `minimizedWord.definitions` is `undefined` if we using Version 1 of the API